collect_app/src/main/java/org/odk/collect/android/tasks/DiskSyncTask.java

```diff
@@ -136,9 +136,15 @@ protected String doInBackground(Void... params) {
                                 uriToUpdate.add(new UriFile(updateUri, sqlFile));
                             }
                         } else {
-                            Log.w(t, "[" + instance
-                                    + "] file referenced by content provider does not exist "
-                                    + sqlFile);
+                           //File not found in sdcard but file path found in database
+                            //probably because the file has been deleted or filename was changed in sdcard
+                            //remove the content from the database
+                            
+                            String id = mCursor.getString(
```

Comment:
I have the same comments with yesterday about this.

We could potentially gather all ids to be deleted here and then bulk delete them in one go after the loop. The drawback of this approach is we need to do slicing if the params going to be more than the maximum variable number in sqlite query (999, defined in application constants).

We could delete it one by one like this and don't deal with the 999 issue, but we're calling one delete statement.